java/src/com/google/template/soy/parsepasses/contextautoesc/Rewriter.java

```diff
@@ -102,7 +103,11 @@
      * Keep track of template nodes so we know which are derived and which aren't.
      */
     @Override protected void visitTemplateNode(TemplateNode templateNode) {
-      Preconditions.checkState(!visitedTemplateNames.contains(templateNode.getTemplateName()));
+      if (visitedTemplateNames.contains(templateNode.getTemplateName())) {
+        throw SoySyntaxExceptionUtils.createWithNode(
+            "was defined previously",
+            templateNode);
+      }
```

Comment:
I agree with the goal of your change, but think that throwing syntax exceptions from this pass is the wrong approach.

Perhaps this check could go in [ResolveNamesVisitor](https://github.com/google/closure-templates/blob/d2fbc7041ae7ee603d70a023369e2b99959e1a47/java/src/com/google/template/soy/passes/ResolveNamesVisitor.java) instead?